MonstieWash/Assets/Scripts/Player/ItemPickup.cs

```diff
@@ -0,0 +1,248 @@
+using System;
+using UnityEngine;
+
+public class ItemPickup : MonoBehaviour
+{
+    [SerializeField] private Transform heldItemsTransform;
+    [SerializeField] private float throwSpeedMultiplier = 10f;
+
+    private ToolSwitcher m_toolSwitcher;
+    private Collider2D m_handCollider;
+    private ContactFilter2D m_contactFilter;
+    private StuckItem m_heldItem;
+    private Transform m_itemInitialParent;
+
+    private float m_initialRotation;
+    private Vector3 m_prevPos = Vector3.zero;
+    private Vector3 m_minimumDir;
+    private bool m_lastWiggleUp = false;
+
+    private bool m_holding { get { return m_heldItem != null; } }
+
+    private void Awake()
+    {
+        m_toolSwitcher = GetComponent<ToolSwitcher>();
+        m_handCollider = GetComponent<Collider2D>();
+        m_contactFilter = new ContactFilter2D();
+        m_contactFilter.SetLayerMask(LayerMask.GetMask("Pickupable"));
+    }
+
+    private void OnEnable()
+    {
+        InputManager.Inputs.OnActivate += Inputs_OnActivate;
+        InputManager.Inputs.OnSwitchTool += Inputs_OnSwitchTool;
+        InputManager.Inputs.OnNavigate += Inputs_OnNavigate;
+    }
+
+    private void OnDisable()
+    {
+        InputManager.Inputs.OnActivate -= Inputs_OnActivate;
+        InputManager.Inputs.OnSwitchTool -= Inputs_OnSwitchTool;
+        InputManager.Inputs.OnNavigate -= Inputs_OnNavigate;
+    }
+
+    private void Inputs_OnActivate()
+    {
+        if (m_toolSwitcher.CurrentToolIndex == -1) //Empty hand
+        {
+            Interact();
+        }
+    }
+
+    private void Inputs_OnSwitchTool(float dirInput)
```

Comment:
Your float parameter here either needs to be an int, or your argument for the OnSwitchTool event (InputManager, line 20) needs to be a float. Otherwise you will get a "No overload matches delegate" error.